ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryMetricsFunctionalTest.java

```diff
@@ -0,0 +1,283 @@
+
+    final MetricName requestDistributionTable = new MetricName(
+        "pull-query-requests-detailed-distribution-90",
+        "_confluent-ksql-pull-query",
+        "Latency distribution - non_windowed-key_lookup-source_node",
+        TABLE_TAGS
+    );
+    final KafkaMetric requestDistributionTableMetric = TestMetricsReporter.METRICS.get(requestDistributionTable);
+
+
+    final MetricName recordsReturnedStream = new MetricName(
+        "pull-query-requests-rows-returned-total",
+        "_confluent-ksql-pull-query",
+        "Number of rows returned - non_windowed_stream-unknown-source_node",
+        STREAMS_TAGS
+    );
+    final KafkaMetric recordsReturnedStreamMetric = TestMetricsReporter.METRICS.get(recordsReturnedStream);
+
+    final MetricName latencyStream = new MetricName(
+        "pull-query-requests-detailed-latency-min",
+        "_confluent-ksql-pull-query",
+        "Min time for a pull query request - non_windowed_stream-unknown-source_node",
+        STREAMS_TAGS
+    );
+    final KafkaMetric latencyStreamMetric = TestMetricsReporter.METRICS.get(latencyStream);
+
+    final MetricName responseSizeStream = new MetricName(
+        "pull-query-requests-detailed-response-size",
+        "_confluent-ksql-pull-query",
+        "Size in bytes of pull query response - non_windowed_stream-unknown-source_node",
+        STREAMS_TAGS
+    );
+    final KafkaMetric responseSizeStreamMetric = TestMetricsReporter.METRICS.get(responseSizeStream);
+
+    final MetricName totalRequestsStream = new MetricName(
+        "pull-query-requests-detailed-total",
+        "_confluent-ksql-pull-query",
+        "Total number of pull query request - non_windowed_stream-unknown-source_node",
+        STREAMS_TAGS
+    );
+    final KafkaMetric totalRequestsStreamMetric = TestMetricsReporter.METRICS.get(totalRequestsStream);
+
+    final MetricName requestDistributionStream = new MetricName(
+        "pull-query-requests-detailed-distribution-90",
+        "_confluent-ksql-pull-query",
+        "Latency distribution - non_windowed_stream-unknown-source_node",
+        TABLE_TAGS
+    );
+    final KafkaMetric requestDistributionStreamMetric = TestMetricsReporter.METRICS.get(requestDistributionStream);
+
+    // When:
+    RestIntegrationTestUtil.makeQueryRequest(
+        REST_APP,
+        "SELECT COUNT, USERID from " + AGG_TABLE + " WHERE USERID='" + AN_AGG_KEY + "';",
+        Optional.empty());
+
+    RestIntegrationTestUtil.makeQueryRequest(
+        REST_APP,
+        "SELECT * from " + PAGE_VIEW_STREAM + " WHERE PAGEID='" + A_STREAM_KEY + "';",
+        Optional.empty());
+
+    for (MetricName metric: TestMetricsReporter.METRICS.keySet()) {
+      if (metric.name().startsWith("pull-query")) {
+        if ((Double)TestMetricsReporter.METRICS.get(metric).metricValue() > 0) {
+          System.out.println(metric + " , " + (Double)TestMetricsReporter.METRICS.get(metric).metricValue());
```

Comment:
This is still a draft PR, but just to make sure it doesn't slip through... we probably don't want this println here.

Reply:
Ooops good catch, thank you!